### ISSUE TITLE
false positive type warning for (+ (byte 32) 1)

### ISSUE THREAD

**Issue by borkdude** [borkdude/clj-kondo#819]
```
$ clj-kondo --lint -  <<< '(+ (byte 32) 1)'
<stdin>:1:4: error: Expected: number, received: byte.
linting took 70ms, errors: 1, warnings: 0
```

Workaround:

```
$ clj-kondo --lint - --config '{:linters {:type-mismatch {:namespaces {clojure.core {byte {:arities {1 {:ret #{:byte :number}}}}}}}}}' <<< '(+ (byte 32) 1)'
linting took 18ms, errors: 0, warnings: 0
```
